### DISCUSSION TITLE
Porting to RTOS

### DISCUSSION THREAD

**Discussion by satzguyz** [nasa/fprime#1395] [Q&A]
Hello

I would like to port Fprime into the FreeRTOS. Is there any guide on the list of abstractions used for Linux OS or reference porting guide on any other RTOS

**Comment by timcanham** [comment]
There was a port underway [here](https://github.com/fprime-community/fprime-freertos), so you can check on the status. In general, the classes in the `Os` folder allow you to specify different implementations depending on the OS. There is also a porting guide [here](https://nasa.github.io/fprime/UsersGuide/dev/porting-guide.html).
